.circleci/config.yml

```diff
@@ -59,9 +59,42 @@ commands:
       - restore_cache:
           name: Restore target cache
           key: target-cache-{{ arch }}-{{ checksum ".circleci/RUST_VERSION" }}-{{ .Environment.CIRCLE_BRANCH }}
+  cargo-check:
+    steps:
+      - run:
+          name: Build
+          command: |
+            cargo check
+          no_output_timeout: 30m
+  cargo-build-test:
+    steps:
+      - run:
+          name: Build tests
+          command: |
+            cargo test --no-run --all
+  cargo-run-test:
+    steps:
+      - run:
+          name: Run tests
+          command: |
+            cargo test --all
+  docker-build-and-publish:
+    steps:
+      - run:
+          name: Build and publish Docker image
+          command: |
+            docker login -u $DOCKER_LOGIN -p $DOCKER_PASSWORD
+            docker build --pull -t cennznet/cennznet:dev -f docker/Dockerfile .
+            docker push cennznet/cennznet:dev
```

Comment:
Is it possible to use `{{ .Environment.CIRCLE_BRANCH }}` as the tag?

Reply:
yeah actually let me add back DOCKER_TAG stuff with branch name